### DISCUSSION TITLE
[Mobile Testing] @wdio/runner: Error: Unable to load spec files quite likely because they rely on `browser` object that is not fully initialized.

### DISCUSSION THREAD

**Discussion by mzuhairiy** [webdriverio/webdriverio#11962] [Q&A]
Hello, I am new at mobile testing using WebDriverIO, I want to try to implement the Page Object Model/Screen Object Model in my project, but there is an error. The error and my code are attached below :

## The error
![image](https://github.com/webdriverio/webdriverio/assets/47267077/7451ff71-13c8-443f-96e2-988abc31ef1d)

## Spec file 
![image](https://github.com/webdriverio/webdriverio/assets/47267077/ed5f2fce-11e0-490e-9999-6a2f3c21763d)

## **Login**
![image](https://github.com/webdriverio/webdriverio/assets/47267077/33e79c5b-95ed-49b9-8ab0-a449d2625850)

## **Left Side Menu**
![image](https://github.com/webdriverio/webdriverio/assets/47267077/36152d58-801c-49b8-9745-cd1b12b3231e)

## **Catalog**
![image](https://github.com/webdriverio/webdriverio/assets/47267077/d50b23b8-ac86-48fc-8fe8-5a4dbf2de488)

## **wdio.conf.js**
![image](https://github.com/webdriverio/webdriverio/assets/47267077/67ba913a-e7b2-4a46-9f92-ea661607947d)

please help me to fix this, thank you

**Comment by christian-bromann** [comment]
@mzuhairiy you error seems to be CJS/ESM related. Your tests which are recognised as CJS modules are trying to import your page objects which are ESM. I recommend to check your `package.json` and set `"type": "module"`.

**Reply by mzuhairiy** [reply]
thank you for the answer but I already set `"type" : "module"` before, here is my `package.json`
![image](https://github.com/webdriverio/webdriverio/assets/47267077/5095b531-0a1c-4c04-9b77-cc0f6bb8f393)

**Reply by christian-bromann** [reply]
Can you provide a reproducible example?

**Reply by mzuhairiy** [reply]
sorry sir what is the meaning of 'reproducible example'? can I provide my github repo so that you can reproduce the error?

**Reply by christian-bromann** [reply]
that is exactly what I meant 😉

**Reply by mzuhairiy** [reply]
all right sir, this is my GitHub repo [https://github.com/mzuhairiy/mobile-testing-wdio-saucelabs.git](https://github.com/mzuhairiy/mobile-testing-wdio-saucelabs.git)
thank you in advance🤝

**Reply by christian-bromann** [reply]
I can't access the repo

**Reply by mzuhairiy** [reply]
https://github.com/mzuhairiy/mobile-testing-wdio-saucelabs.git

**Reply by christian-bromann** [reply]
@mzuhairiy something I didn't recognise before, this type of error:

```
Cannot find module '/private/tmp/mobile-testing-wdio-saucelabs/test/screenObjects/leftSideMenu.screen' 
```

tells you that you are missing the file extension `.js`. In ESM this is required. So better get used to always use a file extension 😉
